infra/lambda_functions/websocket/joindiagram/app.js

```diff
@@ -19,13 +19,51 @@ exports.handler = async (event) => {
       event.requestContext.domainName + "/" + event.requestContext.stage,
   });
 
+  console.log("event body:", body);
+
+  let usersOnDiagramResult;
+  try {
+    usersOnDiagramResult = await ddb
+      .query({
+        TableName: OPEN_DIAGRAMS_TABLE_NAME,
+        KeyConditionExpression: "diagramId = :d",
+        ExpressionAttributeValues: {
+          ":d": body.diagramId,
+        },
+      })
+      .promise();
+  } catch (e) {
+    console.log("Error when querying the users on the open diagram: ", e);
+    return { statusCode: 500, body: e.stack };
+  }
+
+  let newUsersIsMaster = false;
+  if (usersOnDiagramResult.Items.length === 0) {
+    newUsersIsMaster = true;
+    try {
+      await apigwManagementApi
+        .postToConnection({
```

Comment:
again, similar to what I was saying before, if we used dynamodb streams here, this would be a lot neater, and reduce the possibility of having multiple clients thinking that they're master. So we tell dynamodb about the new user and the fact that they are master, and a lambda function is triggered which tells the user that they're master.